File: 6/spec/fizzbuzz_spec.rb
require 'fizzbuzz'

describe 'Fizzbuzz' do
  context 'it should know when it is divsisible' do
    it '3' do
      expect(is_divisible_by_three?(3)).to be true
    end
    it '5' do
    expect(is_divisible_by_five?(5)).to be true
    end
    it '15' do
    expect(is_divisible_by_fifteen?(15)).to be true
    end
  end
  context 'it should know when it is NOT divisible' do
    it '3' do
      expect(is_divisible_by_three?(4)).not_to be true
    end
    it '5' do
    expect(is_divisible_by_five?(7)).not_to be true
    end
    it '15' do
    expect(is_divisible_by_fifteen?(12)).not_to be true
    end
  end
  context 'playing the game' do
    it 'say "Fizz" when divisible by three' do
      expect(fizzbuzz(3)).to eq "Fizz"
    end
    it 'say "Buzz" when divisible by three' do
      expect(fizzbuzz(5)).to eq "Buzz"
    end
    it 'say "FizzBuzz" when divisible by three' do
      expect(fizzbuzz(15)).to eq "FizzBuzz"
    end
    it 'say "number" when divisible by three' do
      expect(fizzbuzz(7)).to eq 7
    end
  end

end

File: javabuzz/11/spec/fizzbuzzSpec.js
describe("Fizzbuzz", function() {
  var fizzbuzz;
  beforeEach(function() {
    fizzbuzz = new Fizzbuzz();
  });
  describe("should know when a number IS divisible by", function() {
    it('3', function() {
      expect(fizzbuzz.isDivisibleByThree(3)).toBe(true);
    });
    it('5', function() {
      expect(fizzbuzz.isDivisibleByFive(5)).toBe(true);
    });
    it('15', function() {
      expect(fizzbuzz.isDivisibleByFifteen(15)).toBe(true);
    });
  });
  describe("should know when a number is NOT divisble by", function() {
    it('three', function() {
      expect(fizzbuzz.isDivisibleByThree(7)).toBe(false);
    });
    it('5', function() {
      expect(fizzbuzz.isDivisibleByFive(4)).toBe(false);
    });
    it('15', function() {
      expect(fizzbuzz.isDivisibleByFifteen(11)).toBe(false);
    });
  });
  describe("when playing should say", function() {
    it('"Fizz"', function() {
      expect(fizzbuzz.game(3)).toEqual("Fizz");
    });
    it('"Buzz"', function() {
      expect(fizzbuzz.game(5)).toEqual("Buzz");
    });
    it('"FizzBuzz"', function() {
      expect(fizzbuzz.game(15)).toEqual("FizzBuzz");
    });
    it('11', function() {
      expect(fizzbuzz.game(11)).toEqual(11);
    });
  });
});
